### ISSUE TITLE
Contractions with straight apostrophes in French

### ISSUE THREAD

**Issue by kysko** [bartosz-antosik/vscode-spellright#568]
Hello, and firstly thanks for this VS Code extension.

The TLDR for this issue is this: contracted articles or adjectives or pronouns with straight apostrophes in French disappear upon correction.

---

For example, when writing in French (or French-Canadian locale in my case), the extension will correctly see `l'etang` as an error, but will correct it with `étang`, rather than `l'étang`: the `l'` disappear.

However:

- if I write with a curly apostrophe, `l’etang` will be (correctly) corrected to `l’étang`, keeping the contracted article;
- `l'étang` is not seen as a mistake, so straight apostrophes seems to be understood upon reading, but not while correcting.

Now, the correct (or most correct) typography in French is the curly apostrophe in these cases, which usually will be automatic in word processors (Word, LO Write, etc), but not so much in plain text editors like VS Code where the straight apostrophe will be inserted. In my case, a tool like `pandoc` will transform those straight apostrophes into curly ones. (And frankly, I prefer it that way.)

If that can help, affected contracted forms to be split from can be:

```
c', d', j', l', m', n', qu', s', t'
```

and also `puisqu'`, `lorsqu'`, `jusqu'`, `quelqu'`

followed by a word to check.

(A weird form like `aujourd'hui` is already correctly recognized as a word in the dictionary, so nothing to do with this.)

Another example: `puisq'alors` will (correctly) be seen as a mistake, and (correctly) corrected to `puisqu'alors`, without curly apostrophe: so it does recognize straight apostrophes.
However, the error `puisqu'alor` will only propose `alors` for a correction, and would forget the `puisqu'` prefix.

Maybe it's just a question of proper splitting, like in the English possessive cases; I hope it's that simple, and not too deep into the OS's backend spell checker.

```
OS: Windows 10
locale: French (Canada)
Document: markdown
```

(Straight apostrophes can occur even in French texts from France. See [here](https://www.france24.com/fr/20161002-pourquoi-il-ny-a-apostrophe-guillemets-nos-claviers) for example, so this is not only a french-canadian keyboard/locale situation.)
